### PR TITLE
174274064 community resolutions

### PR DESCRIPTION
**Pull request by ShaidullinArtur**
#### A brief description of the pull request.

[PT Story 174274064](https://www.pivotaltracker.com/story/show/174274064)

[Design](https://projects.invisionapp.com/d/main#/console/20307299/426705102/preview)

#### Screenshots in case of UI changes

Снимок экрана 2020-08-26 в 18 07 08

# Test plan

#### List of steps to manually test introduced functionality:

- Go to Community page ([Staff like](http://localhost:3000/c/staff/3))
- Check if everything is working correctly

### CODE REVIEW

File: assets/javascripts/discourse/templates/components/dialog-list-item.hbs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{{~topic-link dialog class="raw-link raw-topic-link dialog-list-item__name"}}
```
Comment (DmitryBarskov) [inline assets/javascripts/discourse/templates/components/dialog-list-item.hbs:1]: What is `~`? I found no such operator in Handlebars docs...

**Review by DmitryBarskov — COMMENTED** [review]
image
Now it looks like this for me. Am I doing anything wrong?

**Comment by IlgizGilmanov** [comment]
> image
> Now it looks like this for me. Am I doing anything wrong?

Is it a newly created community?

I'll add conditions so that in the absence of values for the `period` and `voters`, the corresponding fields are not displayed

File: assets/javascripts/discourse/templates/components/dialog-list-item.hbs (hunk 1 above)
Comment (IlgizGilmanov) [reply assets/javascripts/discourse/templates/components/dialog-list-item.hbs:1]: The file `app/assets/javascripts/discourse/app/templates/list/topic-list-item.hbr` contains a description of this operator:
```
  The `~` syntax strip spaces between the elements, making it produce
  `<a class=topic-post-badges>Some text</a><span class=topic-post-badges>`,
  with no space between them.
  This causes the topic-post-badge to be considered the same word as "text"
  at the end of the link, preventing it from line wrapping onto its own line.
```

More info: https://handlebarsjs.com/guide/expressions.html#whitespace-control

**Comment by DmitryBarskov** [comment]
> is it a newly created community?

@IlgizGilmanov no, it's not, but it is a fresh resolution with only 1 vote.

**Comment by IlgizGilmanov** [comment]
> > is it a newly created community?
> 
> @IlgizGilmanov no, it's not, but it is a fresh resolution with only 1 vote.

When I first started working on this task, I had previously created communities that also had a problem with displaying resolutions. Then the problem was that the resolution object was missing the recent_resolution_post field. Accordingly, it was not possible to get information on all fields required in the template.

File: plugin.rb [hunk 2]
```diff
@@ -72,11 +82,164 @@
 
   add_to_serializer(:current_user, :homepage_id) { object.user_option.homepage_id }
 
+  add_to_serializer(:topic_list, :dialogs, false) do
+    object.dialogs.map do |dialog|
```
Comment (Vadim-Kurnatovskiy) [inline plugin.rb]: add check that dialogs presents

**Pull request merged**